pkg/cluster_upgrader/cluster_upgrader.go

```diff
@@ -67,29 +65,38 @@ const (
 // Interface describing the functions of a cluster upgrader.
 //go:generate mockgen -destination=mocks/cluster_upgrader.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/cluster_upgrader ClusterUpgrader
 type ClusterUpgrader interface {
-	UpgradeCluster(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) error
+	UpgradeCluster(upgradeConfig *upgradev1alpha1.UpgradeConfig, logger logr.Logger) error
+}
+
+//go:generate mockgen -destination=mocks/cluster_upgrader.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/cluster_upgrader ClusterUpgraderBuilder
+type ClusterUpgraderBuilder interface {
```

Comment:
Has the mock been regenerated with this added interface? Not seeing it as part of the commit.

Reply:
I will regenerate once I start on fixing the tests. I wanted early feedback before I invest in the test :smile: 
Shall I commit them now so you can review?

Reply:
Oh! No, that's fine, keep on going. :D  This all looks great to me.